pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtGenerator.java

```diff
@@ -70,23 +75,27 @@ public String generate(final U profile) {
 
             // claims
             final JWTClaimsSet claims = builder.build();
+
             // signed
-            final SignedJWT signedJWT = new SignedJWT(new JWSHeader(JWSAlgorithm.HS256), claims);
+            final SignedJWT signedJWT = new SignedJWT(new JWSHeader(JWSAlgorithm.HS256, JOSEObjectType.JWT, null, null, null, null, null, null, null, null, null, null, null), claims);
```

Comment:
From this example: http://connect2id.com/products/nimbus-jose-jwt/examples/jwt-with-rsa-signature, this would be the source code to just sign the token:

```
SignedJWT signedJWT = new SignedJWT(new JWSHeader(JWSAlgorithm.RS256), claims);
signedJWT.sign(signer);
return signedJWT.serialize();
````

So your algorithm looks good except the update of the `SignedJWT` constructor, I don't see the necessity.